src/app/modules/ewt-patron/stake/stake.component.ts

```diff
@@ -66,8 +67,22 @@ export class StakeComponent {
     this.store.dispatch(PoolActions.openWithdrawDialog());
   }
 
-  setAmountValidators(maxAmount: number) {
-    this.amountToStake.setValidators([Validators.min(this.MINIMAL_VALUE), Validators.required, Validators.max(maxAmount)]);
+  setAmountValidators(maxAmount: number, balance: number) {
+    this.amountToStake.setValidators([Validators.min(this.MINIMAL_VALUE), Validators.required, Validators.max(maxAmount), this.valid(balance)]);
+  }
+
+  private valid(balance: number): ValidatorFn {
```

Comment:
I would change this to a more suggestive name

Reply:
haha, forgot about it :D it should be temporary name :D thanks